### ISSUE TITLE
ActivityPub objects include non-ActivityPub references

### ISSUE THREAD

**Issue by koehn** [friendica/friendica#8601]
### Expected behavior
Identifiers in ActivityPub objects returned by Friendica should refer to ActivityPub objects. 

### Actual behavior
The `inReplyTo` fields on objects often contains a reference to a non-activityPub object (e.g., Diaspora). I think for replies to non-AP items this field should not be used. 

I may be wrong. 

### Steps to reproduce the problem
1. `curl -H 'Accept: application/activity+json' https://friendica.mrpetovan.com/objects/735a2029-425e-b321-b930-d73243755806`

2. Examine `context`, `inReplyTo`

### Friendica version you encountered the problem
develop

**Comment by annando** [comment]
That's complicated. We are using AP for out internal communication as well - and we are a cross protocol system. So it is crucial for us to transmit this data as well. This is the reason why we had included these non AP contacts in the followers/followings as well.

**Comment by koehn** [comment]
Yeah, I knew it would be, given the nature of your system. But a consequence is that it makes life a lot harder for interoperability with the rest of the ActivityPub world, including my little server. 

I guess my advice would be to not co-opt interoperability fields for other uses; instead add fields to your own namespace for data that other systems will ignore. 

I also recognize that that's not a simple change to make at this stage of the game.

**Comment by annando** [comment]
Normally such messages shouldn't be transported to AP contacts, so it shouldn't matter. How had you received that message? Had you pulled the outbox?

**Comment by koehn** [comment]
Exactly so.
